File: sdks/sdkwork-mail-sdk/sdkwork-mail-sdk-go/Mailstandard/provider_catalog.go
package Mailstandard

type MailProviderCatalogEntry struct {
    ProviderKey     string
    PluginId        string
    DriverId        string
    DefaultSelected bool
}

type MailProviderCatalog struct{}

const DEFAULT_mail_PROVIDER_KEY = "smtp"

var OFFICIAL_mail_PROVIDERS = []MailProviderCatalogEntry{
    {ProviderKey: "smtp", PluginId: "Mail-smtp", DriverId: "sdkwork-mail-driver-smtp", DefaultSelected: true},
    {ProviderKey: "imap", PluginId: "Mail-imap", DriverId: "sdkwork-mail-driver-imap", DefaultSelected: false},
}

func GetMailProviderByProviderKey(providerKey string) *MailProviderCatalogEntry {
    for index := range OFFICIAL_mail_PROVIDERS {
        if OFFICIAL_mail_PROVIDERS[index].ProviderKey == providerKey {
            return &OFFICIAL_mail_PROVIDERS[index]
        }
    }

    return nil
}

File: sdks/sdkwork-mail-sdk/sdkwork-mail-sdk-rust/src/provider_catalog.rs
#[derive(Clone, Copy)]
#[allow(non_snake_case)]
pub struct MailProviderCatalogEntry {
    pub providerKey: &'static str,
    pub pluginId: &'static str,
    pub driverId: &'static str,
    pub defaultSelected: bool,
}

pub struct MailProviderCatalog;

pub const DEFAULT_mail_PROVIDER_KEY: &str = "smtp";

pub const OFFICIAL_mail_PROVIDERS: [MailProviderCatalogEntry; 2] = [
    MailProviderCatalogEntry { providerKey: "smtp", pluginId: "Mail-smtp", driverId: "sdkwork-mail-driver-smtp", defaultSelected: true },
    MailProviderCatalogEntry { providerKey: "imap", pluginId: "Mail-imap", driverId: "sdkwork-mail-driver-imap", defaultSelected: false },
];

pub fn get_mail_provider_by_provider_key(
    provider_key: &str,
) -> Option<&'static MailProviderCatalogEntry> {
    OFFICIAL_mail_PROVIDERS
        .iter()
        .find(|entry| entry.providerKey == provider_key)
}
